src/main.zig

```diff
@@ -93,14 +93,11 @@ pub fn serialize(comptime T: type, allocator: Allocator, data: T, list: *ArrayLi
             } else {
                 const index = list.items.len;
                 try list.append(0);
-                var length = tlist.items.len;
-                var length_length: u8 = 0;
-                while (length != 0) : (length >>= 8) {
-                    try list.append(@as(u8, @truncate(length)));
-                    length_length += 1;
-                }
-
-                list.items[index] = 247 + length_length;
+                var enc_length_buf: [8]u8 = undefined;
+                std.mem.writeInt(usize, &enc_length_buf, tlist.items.len, .Big);
+                const enc_length = std.mem.trimLeft(u8, &enc_length_buf, &[_]u8{0});
+                try list.appendSlice(enc_length);
+                list.items[index] = 247 + @as(u8, @intCast(enc_length.len));
```

Comment:
Here's the same fix, but for long lists.
The previous code was a bit hairy to solve this case, so I rely more in mainstream std funcs to resolve this.

Reply:
yeah, unfortunately std funcs do not support writing packed values.